src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,88 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 1 << 4;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K,V>[] table;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+        threshold = (int) (DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= threshold) {
+            resize();
+        }
+        int hashOfKey = getHash(key);
+        Node<K, V> newNode = new Node<>(key, value, null);
+        if (table[hashOfKey] == null) {
+            table[hashOfKey] = newNode;
+        } else {
+            Node<K, V> currentNode = table[hashOfKey];
+            while (true) {
+                if (Objects.equals(currentNode.key, newNode.key)) {
+                    currentNode.value = newNode.value;
+                    return;
+                }
+                if (currentNode.next == null) {
+                    currentNode.next = newNode;
+                    break;
+                }
+                currentNode = currentNode.next;
+            }
```

Comment:
using a while loop with a `true` condition may be confusing.

should use concrete conditions, e.g. `while (currentNode != null) {...}`